File: packages/native/ios/Classes/Node/SkityShapeNodes.m
// Licensed under the Apache License Version 2.0 that can be found in the
// LICENSE file in the root directory of this source tree.

#import "SkityShapeNodes.h"

#import <Lynx/LynxComponentRegistry.h>

// Each shape registers a "skity-<name>" Lynx tag (matching src/elements.ts) and
// reports the SVG-style tag_name (rect/circle/...) that SkityRenderer dispatches
// on. isVirtual = YES is inherited from SkityNodeBase.

@implementation SkityRectShadowNode
#if LYNX_LAZY_LOAD
LYNX_LAZY_REGISTER_SHADOW_NODE("gesso-rect")
#else
LYNX_REGISTER_SHADOW_NODE("gesso-rect")
#endif
- (NSString *)skityTagName {
  return @"rect";
}
@end

@implementation SkityCircleShadowNode
#if LYNX_LAZY_LOAD
LYNX_LAZY_REGISTER_SHADOW_NODE("gesso-circle")
#else
LYNX_REGISTER_SHADOW_NODE("gesso-circle")
#endif
- (NSString *)skityTagName {
  return @"circle";
}
@end

@implementation SkityEllipseShadowNode
#if LYNX_LAZY_LOAD
LYNX_LAZY_REGISTER_SHADOW_NODE("gesso-ellipse")
#else
LYNX_REGISTER_SHADOW_NODE("gesso-ellipse")
#endif
- (NSString *)skityTagName {
  return @"ellipse";
}
@end

@implementation SkityLineShadowNode
#if LYNX_LAZY_LOAD
LYNX_LAZY_REGISTER_SHADOW_NODE("gesso-line")
#else
LYNX_REGISTER_SHADOW_NODE("gesso-line")
#endif
- (NSString *)skityTagName {
  return @"line";
}
@end

@implementation SkityPathShadowNode
#if LYNX_LAZY_LOAD
LYNX_LAZY_REGISTER_SHADOW_NODE("gesso-path")
#else
LYNX_REGISTER_SHADOW_NODE("gesso-path")
#endif
- (NSString *)skityTagName {
  return @"path";
}
@end

@implementation SkityPolylineShadowNode
#if LYNX_LAZY_LOAD
LYNX_LAZY_REGISTER_SHADOW_NODE("gesso-polyline")
#else
LYNX_REGISTER_SHADOW_NODE("gesso-polyline")
#endif
- (NSString *)skityTagName {
  return @"polyline";
}
@end

@implementation SkityPolygonShadowNode
#if LYNX_LAZY_LOAD
LYNX_LAZY_REGISTER_SHADOW_NODE("gesso-polygon")
#else
LYNX_REGISTER_SHADOW_NODE("gesso-polygon")
#endif
- (NSString *)skityTagName {
  return @"polygon";
}
@end

@implementation SkityImageShadowNode
#if LYNX_LAZY_LOAD
LYNX_LAZY_REGISTER_SHADOW_NODE("gesso-image")
#else
LYNX_REGISTER_SHADOW_NODE("gesso-image")
#endif
- (NSString *)skityTagName {
  return @"image";
}
@end

@implementation SkityGroupShadowNode
#if LYNX_LAZY_LOAD
LYNX_LAZY_REGISTER_SHADOW_NODE("gesso-group")
#else
LYNX_REGISTER_SHADOW_NODE("gesso-group")
#endif
- (NSString *)skityTagName {
  return @"g";
}
@end

File: packages/native/android/src/main/java/com/skity/graphics/node/SkityShapeNodes.kt
// Licensed under the Apache License Version 2.0 that can be found in the
// LICENSE file in the root directory of this source tree.
package com.skity.graphics.node

/**
 * Virtual shape/group ShadowNodes. Each only declares its skity tag name; all
 * geometry/paint props are inherited from [SkityNodeBase]. The container
 * [SkityCanvasShadowNode] walks the child tree and serializes every node.
 *
 * Mirrors lynx-native-svg's leaf ShadowNodes (RectShadowNode, CircleShadowNode,
 * …) which likewise just set `svgTagName` plus a few geometry setters.
 */

class SkityRectShadowNode : SkityNodeBase() {
  override val skityTagName = "rect"
}

class SkityCircleShadowNode : SkityNodeBase() {
  override val skityTagName = "circle"
}

class SkityEllipseShadowNode : SkityNodeBase() {
  override val skityTagName = "ellipse"
}

class SkityLineShadowNode : SkityNodeBase() {
  override val skityTagName = "line"
}

class SkityPathShadowNode : SkityNodeBase() {
  override val skityTagName = "path"
}

class SkityPolylineShadowNode : SkityNodeBase() {
  override val skityTagName = "polyline"
}

class SkityPolygonShadowNode : SkityNodeBase() {
  override val skityTagName = "polygon"
}

class SkityImageShadowNode : SkityNodeBase() {
  override val skityTagName = "image"
}

class SkityGroupShadowNode : SkityNodeBase() {
  override val skityTagName = "g"
}
